File: go/test/protocol/default_receiver.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

type DefaultReceiver struct {
	Serializer     DefaultSerializer `toml:"serializer"`
	TelemetryTopic *string           `toml:"telemetry-topic"`
	TopicNamespace *string           `toml:"topic-namespace"`
}

func (receiver *DefaultReceiver) GetSerializer() TestCaseSerializer {
	return receiver.Serializer.GetSerializer()
}

func (receiver *DefaultReceiver) GetTelemetryTopic() *string {
	if receiver.TelemetryTopic == nil {
		return nil
	}

	telemetryTopic := *receiver.TelemetryTopic
	return &telemetryTopic
}

func (receiver *DefaultReceiver) GetTopicNamespace() *string {
	if receiver.TopicNamespace == nil {
		return nil
	}

	topicNamespace := *receiver.TopicNamespace
	return &topicNamespace
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/DefaultReceiver.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class DefaultReceiver
    {
        public DefaultReceiver()
        {
            Serializer = new();
        }

        public DefaultSerializer Serializer { get; set; }

        public string? TelemetryTopic { get; set; }

        public string? TopicNamespace { get; set; }
    }
}
